tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -382,6 +388,14 @@ std::vector<std::unique_ptr<TChain>> MakeFriends(const ROOT::TreeUtils::RFriendI
             frChain->Add((thisFriendFileNames[j] + "?#" + thisFriendChainSubNames[j]).c_str(), thisFriendEntries[j]);
          }
       }
+
+      auto &treeIndex = finfo.fTreeIndexInfos[i];
+      if (treeIndex) {
+         auto *copyOfIndex = static_cast<TVirtualIndex *>(treeIndex->Clone());
+         copyOfIndex->SetTree(frChain.get());
+         frChain->SetTreeIndex(copyOfIndex);
+      }
```

Comment:
IIUC now the `fTreeIndexInfos` owns its `TVirtualIndex`es. Why do we need to `Clone` again here?

Reply:
Each friend tree/chain in each task needs its own copy of the thing.